### PR TITLE
adding label to oai-backend in the file docker-compose.yml

### PR DESCRIPTION
**Pull request by Mazztok45**
I added labels to the oai-backend section in the docker-compose.yml file.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -133,6 +133,12 @@ services:
     volumes:
     - ./oaipmh/fiz-oai-backend.properties:/usr/local/tomcat/conf/fiz-oai-backend.properties:ro
     - backend-logs:/usr/local/tomcat/logs
+    labels:
+      - traefik.http.routers.oai-provider.rule=Host(`oai.portal.mardi4nfdi.${TLD}`)
+      - traefik.http.routers.oai-provider.entrypoints=websecure
+      - traefik.http.routers.oai-provider.tls.certResolver=le
+      - traefik.http.middlewares.test-auth.basicauth.users=bWFyZGktdGVzdC11c2VyOnRlc3Q=
```
Comment (physikerwelt) [inline docker-compose.yml:140]: the username is mising

File: docker-compose.yml [hunk 2]
```diff
@@ -133,6 +133,12 @@ services:
...
+
```
Comment (physikerwelt) [inline docker-compose.yml:141]: one needs to connect the auth middleware with the router, e.g,

traefik.http.routers.service-${service-name}.middlewares=${auth-name}

**Review by physikerwelt — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mazztok45) [reply docker-compose.yml:140]: mardi-test-user would be better?

Comment (Mazztok45) [reply docker-compose.yml:140]: the username was encoded with the original form "user:password" and processed with https://mixedanalytics.com/tools/basic-authentication-generator/

[the remark this answers is not in the document]
Comment (physikerwelt) [reply docker-compose.yml:140]: I think it is the auth provider for oai, so maybe oai-auth?

Comment (physikerwelt) [reply docker-compose.yml:140]: This should happen on the client side. For the server side, we need the information that usually would go to .htaccess You can generate it here http://aspirine.org/htpasswd_en.html keep the password private as this repo is public.

File: docker-compose.yml (hunk 2 above)
Comment (Mazztok45) [reply docker-compose.yml:141]: Should I add this line ?

traefik.http.routers.service-${oai-input.portal.mardi4nfdi.${TLD}}.middlewares=${oai-auth}

Comment (physikerwelt) [reply docker-compose.yml:141]: I get the feeling that you don't fully understand what a traefik middelware is, see https://doc.traefik.io/traefik/middlewares/overview/

File: docker-compose.yml (hunk 1 above)
Comment (physikerwelt) [reply docker-compose.yml]: The build fails

https://github.com/MaRDI4NFDI/portal-compose/actions/runs/7492851367/job/20397278707?pr=459#step:4:21

, as the password is not properly escaped, see https://doc.traefik.io/traefik/middlewares/http/basicauth/

**Review by physikerwelt — APPROVED** [review]

**Comment by physikerwelt** [comment]
Let's give it a try.

**Comment by physikerwelt** [comment]
done in #463

**Pull request closed without merging**